File: VB/WebSolution.Module/Objects.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace WebSolution.Module
	Public Class Child
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _name As String
		Public Property Name() As String
			Get
				Return _name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _name, value)
			End Set
		End Property
		Private _master As Master
		<Association("Master-Children")> _
		Public Property Master() As Master
			Get
				Return _master
			End Get
			Set(ByVal value As Master)
				SetPropertyValue("Master", _master, value)
			End Set
		End Property
	End Class
	<DefaultClassOptions, DefaultListViewOptions(MasterDetailMode.ListViewAndDetailView, True, NewItemRowPosition.None)> _
	Public Class Master
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _name As String
		Public Property Name() As String
			Get
				Return _name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _name, value)
			End Set
		End Property

		<Association("Master-Children"), DisplayName("Children")> _
		Public ReadOnly Property Children() As XPCollection(Of Child)
			Get
				Return GetCollection(Of Child)("Children")
			End Get
		End Property
		Private customCollection_Renamed As XPCollection(Of Child)
		<DisplayName("CustomCollection")> _
		Public ReadOnly Property CustomCollection() As XPCollection(Of Child)
			Get
				If customCollection_Renamed Is Nothing Then
					customCollection_Renamed = New XPCollection(Of Child)(Session)
				End If
				Return customCollection_Renamed
			End Get
		End Property
	End Class
End Namespace

File: CS/WebSolution.Module/Objects.cs
using System;
using DevExpress.Xpo;
using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace WebSolution.Module {
    public class Child : BaseObject {
        public Child(Session session) : base(session) { }
        private string _name;
        public string Name {
            get { return _name; }
            set { SetPropertyValue("Name", ref _name, value); }
        }
        private Master _master;
        [Association("Master-Children")]
        public Master Master {
            get { return _master; }
            set { SetPropertyValue("Master", ref _master, value); }
        }
    }
    [DefaultClassOptions]
    [DefaultListViewOptions(MasterDetailMode.ListViewAndDetailView, true, NewItemRowPosition.None)]
    public class Master : BaseObject {
        public Master(Session session) : base(session) { }
        private string _name;
        public string Name {
            get { return _name; }
            set { SetPropertyValue("Name", ref _name, value); }
        }

        [Association("Master-Children")]
        [DisplayName("Children")]
        public XPCollection<Child> Children {
            get {
                return GetCollection<Child>("Children");
            }
        }
        private XPCollection<Child> customCollection;
        [DisplayName("CustomCollection")]
        public XPCollection<Child> CustomCollection {
            get {
                if (customCollection == null)
                    customCollection = new XPCollection<Child>(Session);
                return customCollection;
            }
        }
    }
}